src/main/java/org/lmdbjava/Cursor.java

```diff
@@ -187,15 +192,17 @@ public boolean prev() {
    * Store by cursor.
    *
    * <p>
-   * This function stores key/data pairs into the database. The cursor is
-   * positioned at the new item, or on failure usually near it.
+   * This function stores key/data pairs into the database.
    *
    * @param key key to store
    * @param val data to store
    * @param op  options for this operation
+   * @return true if the value was put, false if MDB_NOOVERWRITE or
+   *     MDB_NODUPDATA were set and the key/value existed already.
    */
-  public void put(final T key, final T val, final PutFlags... op) {
+  public boolean put(final T key, final T val, final PutFlags... op) {
     if (SHOULD_CHECK) {
+      requireNonNull(txn);
```

Comment:
How could `txn` ever become null (it's checked in the constructor, and the only subsequent assignment is in `renew` and the `SHOULD_CHECK` there will pick up any attempt to make it null)?